### ISSUE TITLE
[Core | Python3.10] Stop using `get_event_loop` 

### ISSUE THREAD

**Issue by rkooo567** [ray-project/ray#25025]
### What happened + What you expected to happen

`get_event_loop` will be deprecated and print spammy logs to the driver. We should use `new_event_loop ` and `get_running_loop` instead. https://docs.python.org/3.10/library/asyncio-eventloop.html#asyncio.get_event_loop

### Versions / Dependencies

Ray for Python 3.10

### Reproduction script

pytest -vs python/ray/tests/test_basic.py::test_put_get

**Comment by rkooo567** [comment]
cc @scv119 btw, without fixing this, using ray will almost certainly produce spammy logs like this;

```
(raylet) /Users/sangbincho/work/ray/python/ray/dashboard/agent.py:36: DeprecationWarning: There is no current event loop
(raylet)   aiogrpc.init_grpc_aio()
(raylet) /Users/sangbincho/work/ray/python/ray/dashboard/agent.py:382: DeprecationWarning: There is no current event loop
(raylet)   loop = asyncio.get_event_loop()
(raylet) /Users/sangbincho/work/ray/python/ray/dashboard/agent.py:36: DeprecationWarning: There is no current event loop
(raylet)   aiogrpc.init_grpc_aio()
(raylet) /Users/sangbincho/work/ray/python/ray/dashboard/agent.py:382: DeprecationWarning: There is no current event loop
(raylet)   loop = asyncio.get_event_loop()
```

**Comment by mvashishtha** [comment]
@rkooo567 is it possible to address this some time soon? The messages are fairly long. I don't want the bot to close this issue.

**Comment by mvashishtha** [comment]
@rkooo567 could you please reopen this issue? These messages are really annoying and I think this issue is still relevant.

**Comment by rkooo567** [comment]
@scv119 @rickyyx we should fix this fo 3.10 for ray 2.1
